flink-ml-lib/src/main/java/org/apache/flink/ml/feature/imputer/Imputer.java

```diff
@@ -0,0 +1,333 @@
+                        QuantileSummary summary2 = acc2.get(col).compress();
+                        acc2.put(col, summary2.merge(summary1.compress()));
+                    });
+            return acc2;
+        }
+    }
+
+    /**
+     * A stream operator to compute the most frequent value of all input columns of the input
+     * bounded data stream.
+     */
+    private static class MostFrequentStrategyAggregator
+            implements AggregateFunction<Row, Map<String, Map<Double, Long>>, ImputerModelData> {
+        private final String[] columnNames;
+        private final double missingValue;
+
+        public MostFrequentStrategyAggregator(String[] columnNames, double missingValue) {
+            this.columnNames = columnNames;
+            this.missingValue = missingValue;
+        }
+
+        @Override
+        public Map<String, Map<Double, Long>> createAccumulator() {
+            Map<String, Map<Double, Long>> accumulators = new HashMap<>();
+            Arrays.stream(columnNames).forEach(x -> accumulators.put(x, new HashMap<>()));
+            return accumulators;
+        }
+
+        @Override
+        public Map<String, Map<Double, Long>> add(
+                Row row, Map<String, Map<Double, Long>> accumulators) {
+            accumulators.forEach(
+                    (col, counts) -> {
+                        Double rawValue = (Double) row.getField(col);
+                        boolean shouldBypass =
+                                rawValue == null
+                                        || Double.isNaN(rawValue)
+                                        || rawValue == missingValue;
+                        if (!shouldBypass) {
+                            double value = rawValue;
+                            if (counts.containsKey(value)) {
+                                counts.put(value, counts.get(value) + 1);
+                            } else {
+                                counts.put(value, 1L);
+                            }
+                        }
+                    });
+            return accumulators;
+        }
+
+        @Override
+        public ImputerModelData getResult(Map<String, Map<Double, Long>> map) {
+            long validColumns =
+                    map.entrySet().stream().filter(x -> x.getValue().size() > 0).count();
+            Preconditions.checkState(
+                    validColumns > 0, "The training set is empty or does not contains valid data.");
+
+            Map<String, Double> surrogates = new HashMap<>();
+            map.forEach(
+                    (col, counts) -> {
+                        long maxCnt = Long.MIN_VALUE;
+                        double value = Double.NaN;
+                        for (Map.Entry<Double, Long> entry : counts.entrySet()) {
+                            if (maxCnt < entry.getValue()) {
+                                maxCnt = entry.getValue();
+                                value = entry.getKey();
```

Comment:
Let's guarantee that when there are multiple max counts, the smallest value would be selected.